### PR TITLE
Briefly document interval analysis

### PR DESCRIPTION
**Pull request by xbauch**

### CODE REVIEW

File: src/analyses/README.md [hunk 1]
```diff
@@ -112,6 +112,9 @@ To be documented.
 
 To be documented.
 
-\subsection analyses-interval-analysist Integer interval analysis (interval_analysist) -- both an analysis and a transformation
+\subsection analyses-interval-analysis Integer interval analysis (interval_analysis) -- both an analysis and a transformation
```
Comment (chrisr-diffblue) [inline src/analyses/README.md]: Could make `interval_analysis` here be `\ref interval_analysis`

File: src/analyses/interval_analysis.cpp [hunk 2]
```diff
@@ -7,14 +7,25 @@ Author: Daniel Kroening, kroening@kroening.com
...
 
 #include "interval_domain.h"
 
+/// Instruments all "post-branch" instructions with assertions about variable
+/// intervals. For each such instruction, the function evaluates all variables
+/// referenced within the input goto_function as intervals, transforms these
+/// intervals into expressions and instruments the instruction with their
+/// conjunction.
+/// Example: interval [5,10] (for variable "x") translates into conjunction
+/// x > 5 && x < 10.
```
Comment (chrisr-diffblue) [inline src/analyses/interval_analysis.cpp]: Should that be `x <= 10` ?

File: src/analyses/interval_analysis.h [hunk 3]
```diff
@@ -8,6 +8,9 @@ Author: Daniel Kroening, kroening@kroening.com
 
 /// \file
 /// Interval Analysis
```
Comment (chrisr-diffblue) [inline src/analyses/interval_analysis.h]: Remove this line

File: src/analyses/interval_analysis.cpp [hunk 4]
```diff
@@ -7,14 +7,25 @@ Author: Daniel Kroening, kroening@kroening.com
...
+/// abstract interpretation over interval domain for goto programs. The result
+/// of the analysis is an instrumented program.
 
 #include "interval_analysis.h"
 
 #include <util/find_symbols.h>
 
 #include "interval_domain.h"
 
+/// Instruments all "post-branch" instructions with assertions about variable
```
Comment (chrisr-diffblue) [inline src/analyses/interval_analysis.cpp]: I guess technically we are instrumenting with `assumptions` and not `assertions` ?

File: src/analyses/interval_analysis.cpp (hunk 2 above)
Comment (xbauch) [reply src/analyses/interval_analysis.cpp]: I thought not, but according to `interval_domain.cpp:405` it seems that [5,10] in fact represents `5<=x && x<=10`. Or am I reading it wrong?

File: src/analyses/README.md [hunk 5]
```diff
@@ -112,6 +112,9 @@ To be documented.
 
 To be documented.
 
-\subsection analyses-interval-analysist Integer interval analysis (interval_analysist) -- both an analysis and a transformation
+\subsection analyses-interval-analysis Integer interval analysis (\ref interval_analysis) -- both an analysis and a transformation
 
-To be documented.
+\ref interval_analysis interprets instructions of the input \ref goto_modelt
+over the \ref interval_domaint, evaluates variables for each program location
+(as intervals) and instruments the program with assertions representing the
```
Comment (peterschrammel) [inline src/analyses/README.md]: It seems that it is doing assumptions (the naming in the code is a bit confusing).

File: src/analyses/README.md [hunk 6]
```diff
@@ -112,6 +112,9 @@ To be documented.
 
 To be documented.
 
-\subsection analyses-interval-analysist Integer interval analysis (interval_analysist) -- both an analysis and a transformation
+\subsection analyses-interval-analysis Integer interval analysis (\ref interval_analysis) -- both an analysis and a transformation
 
-To be documented.
+\ref interval_analysis interprets instructions of the input \ref goto_modelt
```
Comment (peterschrammel) [inline src/analyses/README.md]: This ref seems broken

File: src/analyses/interval_analysis.cpp [hunk 7]
```diff
@@ -76,6 +87,9 @@ void instrument_intervals(
   }
 }
 
+/// Initialises the abstract interpretation over interval domain and
+/// instruments instructions using interval assertions.
```
Comment (peterschrammel) [inline src/analyses/interval_analysis.cpp]: assumptions

File: src/analyses/interval_analysis.h [hunk 8]
```diff
@@ -7,7 +7,9 @@ Author: Daniel Kroening, kroening@kroening.com
 \*******************************************************************/
 
 /// \file
-/// Interval Analysis
+/// Interval Analysis -- implements the functionality necessary for performing
```
Comment (peterschrammel) [inline src/analyses/interval_analysis.h]: you don't have to repeat it in both files

File: src/analyses/README.md (hunk 1 above)
Comment (xbauch) [reply src/analyses/README.md]: Links do not seem to work in \subsection.

File: src/analyses/interval_analysis.cpp (hunk 7 above)
Comment (xbauch) [reply src/analyses/interval_analysis.cpp]: fixed.

**Comment by peterschrammel** [comment]
Please squash the commits that fix previous commits of the same PR into the respective commits.

**Comment by xbauch** [comment]
I did some squashing, hopefully without breaking anything too badly.

**Comment by owen-mc-diffblue** [comment]
@peterschrammel Why did you assign me?

**Review by NlightNFotis — APPROVED** [review]
LGTM. Can the two commits be squashed, since the second just looks like it's implementing comment suggestions.

**Comment by xbauch** [comment]
Sure thing.

File: src/analyses/README.md (hunk 1 above)
Comment (peterschrammel) [reply src/analyses/README.md]: then I would remove it the text in the parentheses

**Review by peterschrammel — APPROVED** [review]
Good to go after addressing the last comment.

**Pull request merged**